src/odemis/acq/acq_stream_settings.py

```diff
@@ -0,0 +1,101 @@
+    """
+    # TODO
+    # Any base class required?
+    # check for sorting?,
+    # check for settings values from a preset
+    # change the value and check what is saved
+    # check the order of the settings applied?
+    # todo customtint?
+    file_name = "acquired_stream_mru.json"
+
+    def __init__(self):
+        self.config_data = None
+        self.file_path = os.path.abspath(os.path.join(CONF_PATH, self.file_name))
+        # list of keys to save for storing the stream settings
+        self.setting_keys = ["name", "excitation", "power", "emission", "detExposureTime", "auto_bc_outliers"]
+
+    def read(self):
+        logging.debug("Reading the most recently used acquired settings in JSON")
+        if os.path.exists(self.file_path):
+            try:
+                with open(self.file_path, 'r') as jsonfile:
+                    self.config_data = json.load(jsonfile)
+            except json.decoder.JSONDecodeError as e:
+                print(f"Error decoding JSON: {e}")
+            except Exception as e:
+                print(f"Error in reading acq streams {e}")
+
+    def write(self):
+        logging.debug("Writing the most recently used acquired settings in JSON")
+        if self.config_data:
+            with open(self.file_path, 'w') as jsonfile:
+                json.dump(self.config_data, jsonfile, indent=2)
+        else:
+            logging.warning("No acquired stream data found")
+
+    def update_data(self, new_entries: List[Dict[str, Any]]):
+        """
+        Update the saved acquired stream with the list of new_entries. First element in JSON file
+        is the latest entry while the last element is the oldest acquired stream setting. The file updates
+        its entries based on the new_entries while maintaining a fixed length of entries in the file.
+        :param new_entries: (list of stream controller settings) Each element in the list represent the setting
+         of the acquired stream saved as a dictionary from oldest to latest stream setting
+        """
+        self.read()
+        if self.config_data is None:
+            self.config_data = []
+        data = self.config_data
+        for e in new_entries:
+            # find if the new entry name exists in the saved data
+            index = next((i for i, k in enumerate(data) if k["name"] == e["name"]), None)
+
+            if len(data) < MAX_LENGTH:
+                if index is not None:
+                    data[index] = e
+                else:
+                    data.append(e)
+
+            if len(data) >= MAX_LENGTH:
+                # pop the entry from the list to insert it at the top
+                if index is not None:
+                    data.pop(index)
+                # insert the popped dictionary at the first position
+                data = [e] + data[:MAX_LENGTH - 1]
+                # list of entries should be constant
+                data = data[:MAX_LENGTH]
```

Comment:
Instead of checking the length of the list, I would recommend you add a "last_updated" timestamp to the json. That way you can just add/update the config entries, and then sort by last updated and limit the list to 10 entries.


```python

def get_config_index(config_data: list, name: str) -> int:
    """Get the index of the config entry with the given name
    :param config_data: list of config entries
    :param name: name of config entry
    :return: index of config entry with given name
    """
    try:
        index = list(map(lambda x: x["name"], config_data)).index(name)
    except:
        index = None
    
    return index

def add_and_sort_config_entries(config_data: list, new_entries: list, limit: int = 10) -> list[dict]:
    """Add new entries to config_data and sort by last_updated, limit to 10
    :param config_data: list of config entries
    :param new_entries: list of new config entries
    :param limit: limit of config entries
    :return: sorted list of config entries"""

    # add new entries to config_data
    for entry in new_entries:
        index = get_config_index(config_data, name=entry["name"])
        if index is None:
            config_data.append(entry)
        else:
            config_data[index] = entry

    # sort by last_updated (timestamp)
    sorted_config_data = sorted(config_data, key=lambda x: x["last_updated"], reverse=True)

    return sorted_config_data[:limit] # limit to 10

config_data = add_and_sort_config_entries(config_data, new_entries, limit=10)


```